models/iaf_psc_alpha.cpp

```diff
@@ -240,16 +236,85 @@ iaf_psc_alpha::calibrate()
 
   // these depend on the above. Please do not change the order.
   V_.P30_ = -P_.Tau_ / P_.C_ * numerics::expm1( -h / P_.Tau_ );
-
   V_.P21_ex_ = h * V_.P11_ex_;
-  V_.P31_ex_ = 1 / P_.C_ * ( ( V_.P11_ex_ - V_.P33_ ) / ( -1 / P_.tau_ex_ - -1 / P_.Tau_ )
-                             - h * V_.P11_ex_ ) / ( -1 / P_.Tau_ - -1 / P_.tau_ex_ );
-  V_.P32_ex_ = 1 / P_.C_ * ( V_.P33_ - V_.P11_ex_ ) / ( -1 / P_.Tau_ - -1 / P_.tau_ex_ );
-
   V_.P21_in_ = h * V_.P11_in_;
-  V_.P31_in_ = 1 / P_.C_ * ( ( V_.P11_in_ - V_.P33_ ) / ( -1 / P_.tau_in_ - -1 / P_.Tau_ )
-                             - h * V_.P11_in_ ) / ( -1 / P_.Tau_ - -1 / P_.tau_in_ );
-  V_.P32_in_ = 1 / P_.C_ * ( V_.P33_ - V_.P11_in_ ) / ( -1 / P_.Tau_ - -1 / P_.tau_in_ );
+
+  // exact propagator, linear approximation and singular case for excitatory synaptic event
+
+  const double P32_ex_linear = 1 / 2. / P_.C_ / P_.Tau_ / P_.Tau_ * h * h * ( P_.tau_ex_ - P_.Tau_ )
+    * std::exp( -h / P_.Tau_ );
+  const double P32_ex_singular = h / P_.C_ * V_.P33_;
+  const double P32_ex = 1 / P_.C_ * ( V_.P33_ - V_.P11_ex_ ) / ( -1 / P_.Tau_ - -1 / P_.tau_ex_ );
+  const double P31_ex_linear = h * 2 / 3. * P32_ex_linear;
+  const double P31_ex = 1 / P_.C_ * ( ( V_.P11_ex_ - V_.P33_ ) / ( -1 / P_.tau_ex_ - -1 / P_.Tau_ )
+                                      - h * V_.P11_ex_ ) / ( -1 / P_.Tau_ - -1 / P_.tau_ex_ );
+  const double P31_ex_singular = h * h / 2 / P_.C_ * V_.P33_;
+  const double dev_P31_ex = std::fabs( P31_ex - P31_ex_singular );
+  const double dev_P32_ex = std::fabs( P32_ex - P32_ex_singular );
+  const double dev_norm_ex = std::sqrt( std::pow( dev_P31_ex, 2 ) + std::pow( dev_P32_ex, 2 ) );
+  const double error_linear_ex = std::sqrt(
+    std::pow( std::fabs( P32_ex_linear ), 2 ) + std::pow( std::fabs( P31_ex_linear ), 2 ) );
```

Comment:
Do we really need to compute the roots here? If we didn't, we would just have to replace the factor 2 by 4 in the test below and have exactly the same criterion. 

I also think we could drop the `fabs()`,  since we are squaring anyways.